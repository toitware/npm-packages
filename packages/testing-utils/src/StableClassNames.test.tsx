import { makeStyles } from "@material-ui/core";
import "@testing-library/jest-dom"; // Provides jest dom matchers
import { render } from "@testing-library/react";
import React from "react";
import { StableClassNames } from ".";

const useStyles = makeStyles(() => ({
  test: {
    color: "red",
  },
}));

function Test({ children }: { children: React.ReactNode }): JSX.Element {
  const classes = useStyles();
  return <h1 className={classes.test}>{children}</h1>;
}

test("renders correctly", () => {
  const result = render(
    <StableClassNames>
      <Test>Content</Test>
    </StableClassNames>
  );

  expect(result.container).toMatchSnapshot();
  expect(result.getByText("Content")).toHaveAttribute(
    "class",
    "makeStyles-test" // No numnber at the end.
  );
});
