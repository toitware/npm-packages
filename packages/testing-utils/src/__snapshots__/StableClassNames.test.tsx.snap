// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    class="makeStyles-test"
  >
    Content
  </h1>
</div>
`;
